src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -161,21 +154,17 @@
                     newAttributesDefinitionsDraftsNameMap.get(oldAttributeDefinitionName);
                 return ofNullable(matchingNewAttributeDefinitionDraft)
                     .map(attributeDefinitionDraft -> {
-                        // attribute type is required so if null we let commercetools to throw exception
-                        if (attributeDefinitionDraft.getAttributeType() != null) {
-                            if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
-                                return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
-                            } else {
-                                return Arrays.asList(
-                                    RemoveAttributeDefinition.of(oldAttributeDefinitionName),
-                                    AddAttributeDefinition
-                                        .of(AttributeDefinitionCustomBuilder.of(attributeDefinitionDraft))
-                                );
-                            }
+                        if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
```

Comment:
Discussed for type sync, 
https://github.com/commercetools/commercetools-sync-java/pull/304#discussion_r222698712

Reply:
Yes, you are right about the null checks, I can revert it like the old version with a null check but I'm not sure about the throwing `BuildUpdateActionException` in that method because the method prepares an update action list.

Reply:
@heshamMassoud Could you please check this commit? 
https://github.com/commercetools/commercetools-sync-java/pull/316/commits/465bff8ff3acf71c187ffaf63f4b8fa0ec4619ab